partiql-spi/src/main/kotlin/org/partiql/spi/fn/builtins/FnCollAgg.kt

```diff
@@ -22,9 +22,18 @@ import org.partiql.value.PartiQLValueType
 import org.partiql.value.check
 
 @OptIn(PartiQLValueExperimental::class)
-internal abstract class Fn_COLL_AGG__BAG__ANY : Fn {
+internal abstract class Fn_COLL_AGG__BAG__ANY(
+    name: String,
+    private val isDistinct: Boolean,
+    private val accumulator: () -> Accumulator,
+) : Fn {
 
-    abstract fun getAccumulator(): Agg.Accumulator
+    private fun getAccumulator(): Accumulator = when (isDistinct) {
+        true -> AccumulatorDistinct(accumulator.invoke())
+        false -> accumulator.invoke()
+    }
+
+    override val signature: FnSignature = createSignature(name)
```

Comment:
(unrelated to the changes in this PR) should the return type for some of these `COLL_<AGG>` functions be a more narrow type? E.g. `COLL_SOME` will always return a bool.

Reply:
https://github.com/partiql/partiql-lang-kotlin/issues/1540. Also adding to our internal tracker.